src/commands/install/install.py

```diff
@@ -0,0 +1,48 @@
+from enum import Enum
+from os import path
+from typing import Optional
+from git import InvalidGitRepositoryError, NoSuchPathError, Repo
+from git.objects import Submodule
+from typing_extensions import TypedDict
+from .utils import map_url_to_package_name, verify_repo_url
+
+
+class InstallationErrorEnum(Enum):
+    INCORRECT_URL = "INCORRECT_URL"
+    INVALID_LOCAL_REPOSITORY = "INVALID_LOCAL_REPOSITORY"
+    INVALID_PACKAGE_NAME = "INVALID_PACKAGE_NAME"
+
+
+InstallationResultType = TypedDict(
+    "InstallationResult", {"error": Optional[InstallationErrorEnum]}
```

Comment:
I think we should not return `"error": None` since it might implicate some hidden exception being covered up